internal/internal_task_handlers.go

```diff
@@ -672,23 +674,26 @@ ProcessEvents:
 				return nil, err
 			}
 
-			eventDecisions, err := eventHandler.ProcessEvent(event, isInReplay, isLast)
+			err = eventHandler.ProcessEvent(event, isInReplay, isLast)
 			if err != nil {
 				return nil, err
 			}
+		}
 
-			if eventDecisions != nil {
-				if !isInReplay {
-					w.newDecisions = append(w.newDecisions, eventDecisions...)
-				} else if !skipReplayCheck {
-					replayDecisions = append(replayDecisions, eventDecisions...)
+		// now apply local activity markers
+		for _, m := range markers {
+			if m.MarkerRecordedEventAttributes.GetMarkerName() == localActivityMarkerName {
+				err := eventHandler.ProcessEvent(m, true, false)
+				if err != nil {
+					return nil, err
 				}
 			}
-
-			if w.isWorkflowCompleted {
-				// If workflow is already completed then we can break from processing
-				// further decisions.
-				break ProcessEvents
+		}
+		isReplay := len(reorderedEvents) > 0 && reorderedHistory.IsReplayEvent(reorderedEvents[len(reorderedEvents)-1])
```

Comment:
In Java I create a struct that was return of NextDecisionEvents. This way the isReplay flag and replayDecisions can be part of that struct simplifying this method.